src/zed_utf8.ml

```diff
@@ -224,13 +224,21 @@ let unsafe_extract_prev str ofs =
                   let ch3 = String.unsafe_get str (ofs - 3) in
                   match ch3 with
                     | '\xe0' .. '\xef' ->
-                        (Uchar.of_int (((Char.code ch3 land 0x0f) lsl 12) lor ((Char.code ch2 land 0x3f) lsl 6) lor (Char.code ch1 land 0x3f)), ofs - 3)
+                        let n = (((Char.code ch3 land 0x0f) lsl 12) lor ((Char.code ch2 land 0x3f) lsl 6) lor (Char.code ch1 land 0x3f)) in
+                        if not (Uchar.is_valid n) then
+                          fail str ofs "not a Unicode scalar value"
+                        else
+                          Uchar.of_int n, ofs - 3
```

Comment:
This seems to be duplicated with https://github.com/ocaml-community/zed/pull/51, can you remove this from the current PR?

Reply:
It is a similar problem to #51 in another function. I am going to make this change in another branch.